library/src/scripts/components/mebox/pieces/CompactSearch.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * @author Stéphane LaFlèche <stephane.l@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+import * as React from "react";
+import SearchBar from "@library/components/forms/select/SearchBar";
+import { t } from "@library/application";
+import qs from "qs";
+import apiv2 from "@library/apiv2";
+import Button, { ButtonBaseClass } from "@library/components/forms/Button";
+import classNames from "classnames";
+import { search } from "@library/components/icons/header";
+import { uniqueIDFromPrefix } from "@library/componentIDs";
+import AsyncCreatableSelect from "react-select/lib/AsyncCreatable";
+import SearchOption from "@library/components/search/SearchOption";
+
+export interface ICompactSearchProps {
+    className?: string;
+    placeholder?: string;
+    open: boolean;
+    openSearch: () => void;
```

Comment:
We should name these more similarly to how normal event listener props are named. Eg. `onOpenSearch`, `onCloseSearch()`.